docs/source/ops/DAG-of-services.svg

```diff
@@ -0,0 +1,47 @@
+  <path d="M183.2 237.2 L188.5 243.8 L193.8 237.2 L188.5 250.5 Z" style="stroke:#33322E;fill:#33322E;stroke-dasharray:none;stroke-width:3;"></path>
+  <circle r="4" cx="58.5" cy="25.5" data-name="start" style="stroke:#33322E;fill:#eee8d5;stroke-dasharray:none;stroke-width:3;"></circle>
+  <path d="M58.5 29.5 L58.5 37.5" data-name="start" style="stroke:#33322E;fill:none;stroke-dasharray:none;stroke-width:3;"></path>
+  <path d="M54.5 33.5 L62.5 33.5" data-name="start" style="stroke:#33322E;fill:none;stroke-dasharray:none;stroke-width:3;"></path>
+  <path d="M54.5 41.5 L58.5 37.5 L62.5 41.5" data-name="start" style="stroke:#33322E;fill:none;stroke-dasharray:none;stroke-width:3;"></path>
+  <text x="59" y="59" style="fill: #33322E;font:normal  12pt Helvetica, Helvetica, sans-serif;text-anchor: middle;" data-name="start">start</text>
+  <rect x="46.5" y="108.5" height="31" width="25" data-name="1" style="stroke:#33322E;fill:#eee8d5;stroke-dasharray:none;stroke-width:3;"></rect>
+  <text x="59" y="130" style="fill: #33322E;font:bold  12pt Helvetica, Helvetica, sans-serif;text-anchor: middle;" data-name="1">1</text>
+  <rect x="13.5" y="179.5" height="31" width="25" data-name="2" style="stroke:#33322E;fill:#eee8d5;stroke-dasharray:none;stroke-width:3;"></rect>
+  <text x="26" y="201" style="fill: #33322E;font:bold  12pt Helvetica, Helvetica, sans-serif;text-anchor: middle;" data-name="2">2</text>
+  <rect x="78.5" y="179.5" height="31" width="25" data-name="3" style="stroke:#33322E;fill:#eee8d5;stroke-dasharray:none;stroke-width:3;"></rect>
+  <text x="91" y="201" style="fill: #33322E;font:bold  12pt Helvetica, Helvetica, sans-serif;text-anchor: middle;" data-name="3">3</text>
+  <rect x="143.5" y="179.5" height="31" width="25" data-name="4" style="stroke:#33322E;fill:#eee8d5;stroke-dasharray:none;stroke-width:3;"></rect>
+  <text x="156" y="201" style="fill: #33322E;font:bold  12pt Helvetica, Helvetica, sans-serif;text-anchor: middle;" data-name="4">4</text>
+  <rect x="46.5" y="250.5" height="31" width="25" data-name="5" style="stroke:#33322E;fill:#eee8d5;stroke-dasharray:none;stroke-width:3;"></rect>
+  <text x="59" y="272" style="fill: #33322E;font:bold  12pt Helvetica, Helvetica, sans-serif;text-anchor: middle;" data-name="5">5</text>
+  <rect x="111.5" y="250.5" height="31" width="25" data-name="6" style="stroke:#33322E;fill:#eee8d5;stroke-dasharray:none;stroke-width:3;"></rect>
+  <text x="124" y="272" style="fill: #33322E;font:bold  12pt Helvetica, Helvetica, sans-serif;text-anchor: middle;" data-name="6">6</text>
+  <rect x="176.5" y="250.5" height="31" width="25" data-name="7" style="stroke:#33322E;fill:#eee8d5;stroke-dasharray:none;stroke-width:3;"></rect>
+  <text x="189" y="272" style="fill: #33322E;font:bold  12pt Helvetica, Helvetica, sans-serif;text-anchor: middle;" data-name="7">7</text>
+</svg>
```

Comment:
Process-wise, it would be nice if the svg (or whatever format artifact) were derived at doc build time rather than checked in to the repo.  I don't see any built-in support for nomnoml in sphinx, unfortunately.  There does seem to be an extension which supports nomnoml via the "kroki" service - but adding a remote service as a build dependency seems like a problem of its own.

There is sphinx support for "PlantUML" but that drags in JRE as a dependency.  Maybe that's tolerable but it wouldn't be my first choice either.

Since none of those options seem wonderful, maybe there's nothing actionable here for now - unless this graph can be re-expressed using one of the tools Sphinx does support - graphviz?  If it's not obvious how to do this it probably isn't worth the time right now though - but perhaps add a short note either referencing nomnoml docs for generating this or giving the info directly.

Content-wise, I'm not quite sure how this graph is going to translate into dashboards so the visual might not be accomplishing the intended goal yet, anyway.  Maybe if the numbers were replaced by something else?  Or if the meaning of the numbers were explained somewhere.

Reply:
Hm. I think you are right; then again - this graph isn't going to change, it is only there to complement the "breadth first tree search", to touch up that otherwise text-laden page. Tom Wilkie makes a point of ordering dashboards in this specific way so it is clear to on-call staff who might not have knowledge about the innards of the system which problems lead to which other problems.

But it doesn't really add anything. I'll remove it.

Reply:
D'uh, I only now saw (and remembered) that we aleady have Graphviz support in our docs build stage. Should I put a graphviz graph back in @exarkun ? I could try make it explain more (how to get from what service does what to the order it is listed in the dashboards)

Reply:
I like pictures.  If you think it's going to help the explanation make sense (which I buy) then yea, go for it.

Reply:
I've added some more explanation and Graphviz pictures in eb498726f554412ccb3d21a84286d0dc435cf351.